craft_parts/executor/environment.py

```diff
@@ -19,21 +19,23 @@
 import io
 import logging
 from collections.abc import Iterable
-from typing import Any, cast
+from typing import TYPE_CHECKING, Any, cast
 
 from craft_parts import errors
 from craft_parts.features import Features
-from craft_parts.infos import ProjectInfo, StepInfo
-from craft_parts.parts import Part
-from craft_parts.plugins import Plugin
 from craft_parts.steps import Step
 from craft_parts.utils import os_utils
 
+if TYPE_CHECKING:
+    from craft_parts.infos import ProjectInfo, StepInfo
+    from craft_parts.parts import Part
+    from craft_parts.plugins import Plugin
+
 logger = logging.getLogger(__name__)
 
 
 def generate_step_environment(
-    *, part: Part, plugin: Plugin, step_info: StepInfo
+    *, part: "Part", plugin: "Plugin", step_info: "StepInfo"
```

Comment:
Have you considered a `from __future__ import annotations` to not have to quote imports?  I think that's the more modern approach, if I recall correctly.